### PR TITLE
MM-11399-Manage Admin Roles Using System Console

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation updates for:
- https://mattermost.atlassian.net/browse/MM-28227: New System Roles (Special Admins) - managing via System Console
- https://mattermost.atlassian.net/browse/MM-11399: Add mmctl role reset command

Updated:
- Admin Guide > Onboard Users > Additional System Admin Roles (E20): Added System Console steps for assigning roles and editing privileges
- Admin Guide > Onboard Users > Additional System Admin Roles (E20): Added mmctl for resetting role privileges

### CODE REVIEW

File: source/deployment/admin-roles.rst [hunk 1]
```diff
@@ -2,15 +2,21 @@
...
+From v5.30, System Admins can use the System Console to manage admin roles and admin role privileges.
 
 - **System Manager:** The System Manager has read/write permissions for management areas of the System Console, such as user management and integrations (excluding permissions). This role has read only access to authentication, reporting, and license interfaces.
 - **User Manager:** The User Manager role is able to read/write to all the user management areas (excluding permissions). The authentication interface is read only.
-- **Read Only Admin:** This role is able to access all pages of the System Console but has no write access to any.
+- **Read Only Admin:** This role is able to access all pages of the System Console but has no write access to any pages.
+
+There are two ways to assign roles:
+
+1. Using the System Console. Go to **System Console > User Management > System Roles (Beta)**.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:15]:
```suggestion
1. Using **System Console > User Management > System Roles (Beta)**.
```

File: source/deployment/admin-roles.rst [hunk 2]
```diff
@@ -2,15 +2,21 @@
...
 
-Roles are assigned via the command line, using the `mmctl tool <https://docs.mattermost.com/administration/mmctl-cli-tool.html>`_.
+2. Using the command line, using the `mmctl tool <https://docs.mattermost.com/administration/mmctl-cli-tool.html>`_.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:17]:
```suggestion
2. Using the `mmctl tool <https://docs.mattermost.com/administration/mmctl-cli-tool.html>`_.
```

File: source/deployment/admin-roles.rst [hunk 3]
```diff
@@ -2,15 +2,21 @@
...
 
-Each role has a set of default permissions, and when a user is assigned a role, they are able to access the System Console. The items that they can view depend on the role they've been assigned.
+Each role has a set of default permissions. When a user is assigned a role, they are able to access the System Console. The items that they can view depend on the role they've been assigned.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:19]:
```suggestion
When a user is assigned a role, they have access the System Console. Each role has a set of default permissions. The items that they can view depend on the role they've been assigned.
```

File: source/deployment/admin-roles.rst [hunk 4]
```diff
@@ -45,51 +51,119 @@ Each role has a set of default permissions, and when a user is assigned a role,
...
 
 The format of the mmctl command is:
 
 ``mmctl permissions role assign [role_name] [username...]``
 
 **To grant the System Manager role to a single user called Bob Smith:**
 
+1. Go to **System Console > User Management > System Roles (Beta)** and select the **System Manager** role.
+
+2. Under **Assigned People**, click **Add People**.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:66]:
```suggestion
2. Under **Assigned People**, choose **Add People**.
```

File: source/deployment/admin-roles.rst [hunk 5]
```diff
@@ -45,51 +51,119 @@ Each role has a set of default permissions, and when a user is assigned a role,
...
+
+The format of the mmctl command is:
+
 ``mmctl permissions role assign system_manager bob-smith``
 
 **To grant the User Manager role to two users, Bob Smith and Sue Clark:**
 
+1. Go to **System Console > User Management > System Roles (Beta)** and select the **User Manager** role.
+
+2. Under **Assigned People**, click **Add People**.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:78]:
```suggestion
2. Under **Assigned People**, choose **Add People**.
```

File: source/deployment/admin-roles.rst [hunk 6]
```diff
@@ -45,51 +51,119 @@ Each role has a set of default permissions, and when a user is assigned a role,
...
+
+3. Search for and select ``Bob Smith`` and ``Sue Clark``, and then click **Add** to grant the User Manager role to those users.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:80]:
```suggestion
3. Search for and select **Bob Smith** and **Sue Clark** then choose **Add** to grant the User Manager role to those users.
```

File: source/deployment/admin-roles.rst [hunk 7]
```diff
@@ -45,51 +51,119 @@ Each role has a set of default permissions, and when a user is assigned a role,
...
+
+The format of the mmctl command is:
+
 ``mmctl permissions role assign system_user_manager bob-smith sue-clark``
 
 **To grant the Read Only Admin role to two users, Bob Smith and Sue Clark:**
 
+1. Go to **System Console > User Management > System Roles (Beta)** and select the **Viewer** role.
+
+2. Under **Assigned People**, click **Add People**.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:90]:
```suggestion
2. Under **Assigned People**, select **Add People**.
```

File: source/deployment/admin-roles.rst [hunk 8]
```diff
@@ -45,51 +51,119 @@ Each role has a set of default permissions, and when a user is assigned a role,
...
+
+3. Search for and select ``Bob Smith`` and ``Sue Clark``, and then click **Add** to grant the Viewer role to those users.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:92]:
```suggestion
3. Search for and select **Bob Smith** and **Sue Clark**, then select **Add** to grant the Viewer role to those users.
```

Comment (justinegeffen) [reply source/deployment/admin-roles.rst:92]: In this line, based on our style guide, Viewer should be *Viewer* as it's a role/user. I just realized that which is why it's not formatted in that way. Thoughts on changing the formatting in this doc? Examples of this formatting are in the installation guide (e.g., https://docs.mattermost.com/install/install-ubuntu-2004.html), so it is used in the docs albeit inconsistently.

File: source/deployment/admin-roles.rst [hunk 9]
```diff
@@ -45,51 +51,119 @@ Each role has a set of default permissions, and when a user is assigned a role,
...
+
+The format of the mmctl command is:
+
 ``mmctl permissions role assign system_read_only_admin bob-smith sue-clark``
 
 **To remove the System Manager role from a single user called Bob Smith:**
 
+1. Go to **System Console > User Management > System Roles (Beta)** and select the **Viewer** role.
+
+2. Under **Assigned People**, search for ``Bob Smith``, and then select **Remove**.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:102]:
```suggestion
2. Under **Assigned People**, search for **Bob Smith**, and then select **Remove**.
```

Comment (justinegeffen) [reply source/deployment/admin-roles.rst:102]: Actually, based on the above, "Bob Smith" would be italicized too.

File: source/deployment/admin-roles.rst [hunk 10]
```diff
@@ -45,51 +51,119 @@ Each role has a set of default permissions, and when a user is assigned a role,
...
+
+The format of the mmctl command is:
+
 ``mmctl permissions remove system_user_manager sysconsole_read_authentication``
 
 **To remove write access to the Authentication section of the System Console for all users with the User Manager role:**
 
+1. Go to **System Console > User Management > System Roles (Beta)** and select the **User Manager** role.
+
+2. Under **Privileges > Authentication**, select **No access**, and then click **Save**.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:149]:
```suggestion
2. Under **Privileges > Authentication** select **No access**, then choose **Save**.
```

File: source/deployment/admin-roles.rst [hunk 11]
```diff
@@ -45,51 +51,119 @@ Each role has a set of default permissions, and when a user is assigned a role,
...
 
 ``mmctl permissions role assign [role_name] [username...]``
 
 **To grant the System Manager role to a single user called Bob Smith:**
 
+1. Go to **System Console > User Management > System Roles (Beta)** and select the **System Manager** role.
+
+2. Under **Assigned People**, choose **Add People**.
+
+3. Search for and select ``Bob Smith``, and then click **Add** to grant the System Manager role to that user.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:68]:
```suggestion
3. Search for and select ``Bob Smith`` and choose **Add** to grant the System Manager role to that user.
```

File: source/deployment/admin-roles.rst [hunk 12]
```diff
@@ -45,51 +51,119 @@ Each role has a set of default permissions, and when a user is assigned a role,
...
+
 ``mmctl permissions role assign system_manager bob-smith``
 
 **To grant the User Manager role to two users, Bob Smith and Sue Clark:**
 
+1. Go to **System Console > User Management > System Roles (Beta)** and select the **User Manager** role.
+
+2. Under **Assigned People**, choose **Add People**.
+
+3. Search for and select **Bob Smith** and **Sue Clark** then choose **Add** to grant the User Manager role to those users.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:80]:
```suggestion
3. Search for and select **Bob Smith** and **Sue Clark** and choose **Add** to grant the User Manager role to those users.
```

File: source/deployment/admin-roles.rst [hunk 13]
```diff
@@ -45,51 +51,119 @@ Each role has a set of default permissions, and when a user is assigned a role,
...
+
+The format of the mmctl command is:
+
 ``mmctl permissions role assign system_read_only_admin bob-smith sue-clark``
 
 **To remove the System Manager role from a single user called Bob Smith:**
 
+1. Go to **System Console > User Management > System Roles (Beta)** and select the **Viewer** role.
+
+2. Under **Assigned People**, search for **Bob Smith**, and then select **Remove**.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:102]:
```suggestion
2. Under **Assigned People** search for **Bob Smith**, and select **Remove**.
```

File: source/deployment/admin-roles.rst [hunk 14]
```diff
@@ -45,51 +51,119 @@ Each role has a set of default permissions, and when a user is assigned a role,
...
 
 The format of the mmctl command is:
 
 ``mmctl permissions add [role_name] [permission...]``
 
 **To grant write access to the Authentication section of the System Console for all users with the User Manager role:**
 
+1. Go to **System Console > User Management > System Roles (Beta)** and select the **User Manager** role.
+
+2. Under **Privileges > Authentication**, select **Can edit**, and then click **Save**.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:129]:
```suggestion
2. Under **Privileges > Authentication**, select **Can edit**, and then **Save**.
```

File: source/deployment/admin-roles.rst [hunk 15]
```diff
@@ -45,51 +51,119 @@ Each role has a set of default permissions, and when a user is assigned a role,
...
+
+The format of the mmctl command is:
+
 ``mmctl permissions add system_user_manager sysconsole_write_authentication``
 
 **To grant read only access to the Authentication section of the System Console for all users with the User Manager role:**
 
+1. Go to **System Console > User Management > System Roles (Beta)** and select the **User Manager** role.
+
+2. Under **Privileges > Authentication**, select **Read only**, and then click **Save**.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:139]:
```suggestion
2. Under **Privileges > Authentication**, select **Read only**, and then **Save**.
```

**Comment by justinegeffen** [comment]
@cwarnermm, we chatted a bit offline regarding some of my suggestions. Let me know whether the new ones make sense or should be tabled for the docs 2.0 kickoff discussion.

**Comment by cwarnermm** [comment]
Thanks so much @justinegeffen! Your suggestions are excellent and I've aimed to apply them consistently throughout.

**Comment by wiersgallak** [comment]
@cwarnermm I like the inline placement of the cli command with the action the user is trying to complete.  This is a great way to address the two different ways of assigning and removing roles. When reading through the doc, the CLI portion stands out to me with a bit more emphasis or as a new statement, not as an alternative to the UI. Wondering if adding some additional indentation or adding two headings such as "In the System Console" "With the CLI" or even adding the CLI as a part of the numbered list would help give it the same weight as the UI components.

**Comment by cwarnermm** [comment]
Great feedback, @wiersgallak! I agree that giving more equal weight to both the UI components and the CLI components is needed. Thanks!

File: source/deployment/admin-roles.rst [hunk 16]
```diff
@@ -2,15 +2,21 @@
 Additional System Admin Roles (E20)
 ====================================
 
-From v5.28, System Admins can assign roles to members, allowing them access only to designated areas of the System Console. This allows other members of your organization to perform certain administrative tasks without providing access to all system administrative capabilities.
+From v5.28, System Admins can use the command line to assign roles to members that permit admin access to designated areas of the System Console. This allows other members of your organization to perform certain administrative tasks without providing access to all system administrative capabilities.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:5]:
```suggestion
From v5.28, System Admins can use the command line to assign roles that permit members to have admin access to specified areas of the System Console. This allows members of your organization to perform certain administrative tasks without providing them with access to all system administrative capabilities.
```

File: source/deployment/admin-roles.rst [hunk 17]
```diff
@@ -2,15 +2,21 @@
 Additional System Admin Roles (E20)
 ====================================
 
-From v5.28, System Admins can assign roles to members, allowing them access only to designated areas of the System Console. This allows other members of your organization to perform certain administrative tasks without providing access to all system administrative capabilities.
+From v5.28, System Admins can use the command line to assign roles that permit members to have admin access to specified areas of the System Console. This allows members of your organization to perform certain administrative tasks without providing them with access to all system administrative capabilities.
+
+From v5.30, System Admins can use the System Console to manage admin roles and admin role privileges.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:7]:
```suggestion
From v5.30, System Admins can also use the System Console to manage additional System Admin roles and privileges.
```

File: source/deployment/admin-roles.rst [hunk 18]
```diff
@@ -2,15 +2,21 @@
...
+
+There are two ways to assign roles:
 
-Roles are assigned via the command line, using the `mmctl tool <https://docs.mattermost.com/administration/mmctl-cli-tool.html>`_.
+1. In the System Console under **User Management > System Roles (Beta)**.
 
-Each role has a set of default permissions, and when a user is assigned a role, they are able to access the System Console. The items that they can view depend on the role they've been assigned.
+2. Using the `mmctl tool <https://docs.mattermost.com/administration/mmctl-cli-tool.html>`_.
+
+When a user is assigned a role, they have access the System Console. Each role has a set of default permissions. The items that they can view depend on the role they've been assigned.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:19]:
```suggestion
When a user is assigned a role, they have access the System Console. As each role has a different set of default permissions the items that they can view depend on the role they've been assigned.
```

File: source/deployment/admin-roles.rst [hunk 19]
```diff
@@ -45,50 +51,154 @@ Each role has a set of default permissions, and when a user is assigned a role,
 Assigning Admin Roles
 ---------------------
 
-System Admins can assign roles and modify privileges using the mmctl tool. This can be done either locally or remotely.
+System Admins can assign roles using the System Console or the mmctl tool. This can be done either locally or remotely.
 
-The format of the mmctl command is:
+  **In the System Console**
+
+  1. Go to **System Console > User Management > System Roles (Beta) > Assigned People**.
```
Comment (justinegeffen) [inline source/deployment/admin-roles.rst:58]:
```suggestion
- Go to **System Console > User Management > System Roles (Beta) > Assigned People**. 
```

**Review by justinegeffen — APPROVED** [review]

**Review by mkraft — APPROVED** [review]

**Pull request merged**